src/fn/watch-project.js

```diff
@@ -0,0 +1,177 @@
+                        });
+                        continue;
+                    }
+                    if (fileName.match(formXLSRegex)) {
+                        eventQueue.enqueue(() => {
+                            return convertForms(environment.pathToProject, 'app', {
+                                enketo: true,
+                                forms: [fileName.split('.')[0]],
+                                transformer: xml => xml.replace('</instance>', '</instance>\n      <instance id="contact-summary"/>'),
+                            }).then(() => Promise.resolve(fileName));
+                        });
+                        continue;
+                    }
+                    const formMediaDir = parsedPath.dir.replace(appFormsPath, '');
+                    if (formMediaDir && formMediaDir.match(formMediaRegex)) {
+                        eventQueue.enqueue(() => {
+                            return uploadForms(environment.pathToProject, 'app', {
+                                forms: [formMediaDir.split('-')[0]],
+                            }).then(() => Promise.resolve(fileName));
+                        });
+                        continue;
+                    }
+                    continue;
+                }
+
+                if (parsedPath.dir === contactFormsPath) {
+                    const fileName = parsedPath.base;
+                    if (fileName.match(formXLSRegex)) {
+                        eventQueue.enqueue(() => {
+                            return convertContactForm(environment.pathToProject, [fileName.split('.')[0]])
+                                .then(() => Promise.resolve(fileName));
+                        });
+                        continue;
+                    }
+                    if (fileName.match(formXMLRegex)) {
+                        eventQueue.enqueue(() => {
+                            return uploadForms(environment.pathToProject, 'contact', {
+                                id_prefix: 'contact:',
+                                forms: [fileName.split('.')[0]],
+                                default_context: { person: false, place: false },
+                            }).then(() => Promise.resolve(fileName));
+                        });
+                        continue;
+                    }
+                    continue;
+                }
+
+                if (parsedPath.dir === path.join(environment.pathToProject, 'translations')) {
+                    const fileName = parsedPath.base;
+                    if (fileName.match(/messages-[\w]+\.properties/)) {
+                        await uploadCustomTranslations();
+                        if (callback) callback(fileName);
+                        continue;
+                    }
+                    continue;
+                }
+
+                if ((parsedPath.dir === environment.pathToProject && parsedPath.base !== 'resources.json')
```

Comment:
this condition is very hard to read. Please store it in a variable. It would also be useful if `compileAppSettings` would export the relevant paths, or even the path matcher altogether.